### DISCUSSION TITLE
Any way to determine the Controlled Load tariff from the API?

### DISCUSSION THREAD

**Discussion by GrahamLea** [amberelectric/public-api#97] [Q&A] [answered]
Is there a way to determine which Controlled Load tariff is being applied to a channel from the API? 

I have a Channel ID of `E3` and a channelType of `controlledLoad`.
My DNSP has tariffs for Controlled Load 1 and 2, and I'm not sure which one I have.

Is there a way to read out of the API which Controlled Load tariff I'm on? (Apart from deducing it from the prices.) Is it encoded in the `tariffInformation` somehow?

**Comment by madpilot** [comment, marked as answer]
It's not.

Tariff codes are very slow moving so there isn't a lot of advantage to increasing the size of the payload to include them.

I'm OOO at the moment, but ping info@amber.com.au and they can look up you controlled load tariff code for you.

**Comment by GrahamLea** [comment]
Thanks! I'm pretty sure I can figure out which tariff I'm on from the prices. But I was thinking about building a generic script that would work for any site. I guess I'll have to make the controlled load tariff a user input rather than figuring it out.

**Comment by GrahamLea** [comment]
For anyone else looking for a quick way to figure this out for yourself, find your DNSP's tariff list (e.g. [here's AusGrid's for 2021](https://www.ausgrid.com.au/-/media/Documents/Regulation/Pricing/PList/AUSGRID-NETWORK-PRICE-LIST-FY2021-22.pdf)), and match up the cheaper of the Daily Supply Charges on your Amber bill with the Network Access Charge column for one of the Controlled Load tariffs in the tariff list.

**Comment by madpilot** [comment]
Question (call it API product research): What are you trying to achieve here?

**Reply by GrahamLea** [reply]
It's a part of what I was asking about [over here](https://github.com/amberelectric/public-api/discussions/50#discussioncomment-2224561), i.e. re-creating my bill from API data, so that I can test out alternative tariffs on historical data.

**Reply by GrahamLea** [reply]
At the moment I'm solving this particular part of the program with some config like this:
```
  tariffsByChannelType: {
    "general": "ausgrid/2021/EA025_Residential_ToU.json5",
    "controlledLoad": "ausgrid/2021/EA030_Controlled_Load_1.json5",
  }
```

**Comment by madpilot** [comment]
You can get the tariff code from the API now: https://github.com/amberelectric/public-api/discussions/100

**Reply by GrahamLea** [reply]
Thanks. Looks great! (OpenAPI / docs get updated on a different schedule to the implementation?)

**Reply by madpilot** [reply]
No, try hard refreshing.

**Reply by madpilot** [reply]
Actually, something went wrong - it didn't update automatically. Will fix it shortly.

**Reply by madpilot** [reply]
Caching. It's always caching. Fixed now.
